subprojects/internal-instrumentation-processor/src/main/java/org/gradle/internal/instrumentation/processor/AbstractInstrumentationProcessor.java

```diff
@@ -104,6 +105,8 @@ private void collectAndProcessRequests(Collection<? extends Element> annotatedEl
     @Nonnull
     private static List<ExecutableElement> getExecutableElementsFromAnnotatedElements(Collection<? extends Element> annotatedClassElements) {
         return annotatedClassElements.stream()
+            // Ensure that the type elements have a stable order, as the annotation processing engine does not guarantee it
+            .sorted(Comparator.comparing(AbstractInstrumentationProcessor::elementQualifiedNameIfPresent))
```

Comment:
❓ Wouldn't it be better to sort it after it has been filtered? Or even better, after all elements are known to `ExecuteElement` instances only?